### DISCUSSION TITLE
No "wavelength" option in "dolphin config"

### DISCUSSION THREAD

**Discussion by Shima-shoki** [isce-framework/dolphin#511] [Q&A] [answered]
Hi, I've got a kind advice to specify "wavelength" option to get the final output in a meter unit #506. However, I haven't been able to find it in the "dolphin config" options as shown below. What is an appropriate way of specifying the Sentinel-1 wavelength (= 0.055 m) in the config parameter setting...? Any warm advice is very much appreciated!

```python
usage: dolphin config [-h] [--print-empty] [-o OUTFILE] [--slc-files [SLC_FILES ...]]
                      [--keep-paths-relative] [-sds SUBDATASET]
                      [--amplitude-mean-files [AMPLITUDE_MEAN_FILES ...]]
                      [--amplitude-dispersion-files [AMPLITUDE_DISPERSION_FILES ...]]
                      [--mask-file MASK_FILE] [-ms MINISTACK_SIZE] [--use-evd] [-hw X Y]
                      [--shp-method {glrt,ks,rect}] [--baseline-lag BASELINE_LAG]
                      [--amp-dispersion-threshold AMP_DISPERSION_THRESHOLD] [--no-unwrap]
                      [--unwrap-method {snaphu,icu,phass,spurt,whirlwind}]
                      [-t row_tiles col_tiles] [--downsample-factor row_downsample col_downsample]
                      [--n-parallel-unwrap N_PARALLEL_UNWRAP] [--run-goldstein]
                      [--run-interpolation] [--troposphere-files [TROPOSPHERE_FILES ...]]
                      [--tropo-date-fmt TROPO_DATE_FMT] [--tropo-package TROPO_PACKAGE]
                      [--tropo-model {ECMWF,ERA5,HRES,ERAINT,ERAI,MERRA,NARR,HRRR,GMAO}]
                      [--tropo-delay_type {wet,dry,hydrostatic,comb}]
                      [--ionosphere-files [IONOSPHERE_FILES ...]]
                      [--geometry-files [GEOMETRY_FILES ...]] [--dem-file DEM_FILE]
                      [--single-update] [-s X Y] [--output-bounds left bottom right top]
                      [--log-file LOG_FILE] [--work-directory WORK_DIRECTORY] [--enable-gpu]
                      [--block-shape BLOCK_SHAPE BLOCK_SHAPE]
                      [--n-parallel-bursts N_PARALLEL_BURSTS]
                      [--threads-per-worker THREADS_PER_WORKER]
```

**Comment by Shima-shoki** [comment]
I did find the "wavelength" parameter in the generated .yaml file, so I can also add the value after generating the configuration file, but if there's a way to specify it during the parameter generation step, I really appreciate to know that!
![image](https://github.com/user-attachments/assets/a0a080af-fdfb-42ce-ad57-d9ec6b57f268)

**Reply by scottstanie** [reply, marked as answer]
Yes sorry I didn't specify that- right now the `dolphin config` command line tool has a subset of all possible options because the available options have grown large.

I'm looking into ways to keep the CLI in sync with the actual config object. Right now, the best way is to do what you did- edit the text file, either after specify the options you want that are available via CLI, or with `dolphin config --print-empty` to get a yaml file with all defaults filled in that you can edit in a text editor.

**Reply by Shima-shoki** [reply]
@scottstanie 
>Right now, the best way is to do what you did- edit the text file, either after specify the options you want that are available via CLI, or with dolphin config --print-empty to get a yaml file with all defaults filled in that you can edit in a text editor.

Okay, well noted!! For now I will just edit the .yaml file with the wavelength parameter :)

**Reply by Shima-shoki** [reply]
A small update of the discussion: After adding wavelength (0.055) of the Sentinel-1 (unit: m), the output now looks to be calculated in meters. Thank you very much for the help!
![image](https://github.com/user-attachments/assets/288809c8-3820-4dee-81d3-ed8f6e12d59a)
